common/src/main/java/com/wynntils/screens/maps/widgets/PoiManagerWidget.java

```diff
@@ -0,0 +1,180 @@
+    private Button upButton;
+    private Button downButton;
+    private int row;
+    private CustomColor color;
+    private int spacingMultiplier = 20;
+    private PoiManagementScreen managementScreen;
+    private List<CustomPoi> pois;
+
+    public PoiManagerWidget(
+            float x, float y, int width, int height, CustomPoi poi, int row, PoiManagementScreen managementScreen) {
+        super((int) x, (int) y, width, height, Component.literal(poi.getName()));
+        this.poi = poi;
+        this.row = row;
+        this.managementScreen = managementScreen;
+
+        pois = Managers.Feature.getFeatureInstance(MapFeature.class).customPois.get();
+
+        color = CustomColor.fromInt(0xFFFFFF);
+
+        if (poi.getVisibility() == CustomPoi.Visibility.HIDDEN) {
+            color = CustomColor.fromInt(0x636363);
+        }
+
+        this.editButton = new Button.Builder(
+                Component.translatable("screens.wynntils.poiManagementGui.edit"),
+                (button) -> McUtils.mc().setScreen(PoiCreationScreen.create(managementScreen, poi)))
+                .pos(this.width/2 + 85 + 20, 54 + spacingMultiplier * row)
+                .size(40, 20)
+                .build();
+
+        this.deleteButton = new Button.Builder(
+                Component.translatable("screens.wynntils.poiManagementGui.delete"),
+                (button) -> {
+                    Managers.Feature.getFeatureInstance(MapFeature.class)
+                            .customPois
+                            .get()
+                            .remove(poi);
+                    Managers.Config.saveConfig();
+                    managementScreen.populatePois();
+                })
+                .pos(this.width/2 + 130 + 20, 54 + spacingMultiplier * row)
+                .size(40, 20)
+                .build();
+
+        this.upButton = new Button.Builder(
+                Component.literal("\u2303"),
+                (button) -> updateIndex(-1))
+                .pos(this.width/2 + 172 + 20, 54 + spacingMultiplier * row)
+                .size(9, 9)
+                .build();
+
+        this.downButton = new Button.Builder(
+                Component.literal("\u2304"),
+                (button) -> updateIndex(1))
+                .pos(this.width/2 + 172 + 20, 54 + spacingMultiplier * row + 9)
+                .size(9, 9)
+                .build();
+
+        if (pois.indexOf(poi) == 0) {
+            upButton.active = false;
+        } else if (pois.indexOf(poi) == (pois.size() - 1)) {
+            downButton.active = false;
+        }
+    }
+
+    @Override
+    public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        MultiBufferSource.BufferSource bufferSource =
+                McUtils.mc().renderBuffers().bufferSource();
+
+        float centreZ = 64 + spacingMultiplier * row;
+
+        poi.renderAt(poseStack, bufferSource, this.width / 2f - 151, centreZ, false, 1f, 1f);
```

Comment:
When there is only 1 widget being rendered on the page this doesn't display the icon.

Reply:
Huh? This comment seems out of place. What was the purpose of it?

Reply:
Sorry let me be more clear. When a page is only displaying 1 poi, so the final page, the icon for that poi doesn't render. This code is definetly being ran as the name and all other information about the poi render but for some reason the icon doesn't.
![2023-03-30_13 05 37](https://user-images.githubusercontent.com/8337467/228830815-6a062280-74fc-4e65-a2d0-c0d4274c296d.png)

Reply:
That sounds like it could be it, I'll give that a try later

Reply:
Well I know that's not the problem but I do know what is. It happens when all icons on the same page are the same type.
![2023-03-30_18 35 21](https://user-images.githubusercontent.com/8337467/228919415-ffd3c6fd-8864-45a1-a076-1444caf4fdc8.png)

Reply:
There might be a problem with the poi.renderAt() method. Hm, interesting...

Reply:
I don't think you can use poi.renderAt() here. It is not supposed to work like that. For instance, it takes the mapZoom value and checks the transparency of the icon at that zoom level.

Reply:
You are hard-coding a zoom level of 1f, I guess T3 loot chests are invisible at that zoom level...

Reply:
> There might be a problem with the poi.renderAt() method. Hm, interesting...

I suggested that.. We used it at world distance feature. Maybe it works differently..

Reply:
I'm not sure what a good solution would be. Either add a renderForGui() in Poi.java, or if your pois are only IconPoi's, call getIcon() and render it yourself.

Reply:
> I'm not sure what a good solution would be. Either add a renderForGui() in Poi.java, or if your pois are only IconPoi's, call getIcon() and render it yourself.

You have to handle colors, so method in IconPoi.

Reply:
For now I've just changed it back to using the same icon rendering method that PoiCreationScreen uses. I did try making a renderForGui in IconPoi where the alpha was always 1 and setting z to 1 for all pois but it just displayed exactly the same as renderAt

Reply:
Looks good in-game now.